src/lib/quoter.js

```diff
@@ -52,9 +52,14 @@ class Quoter {
     const hop = this._findBestHopForAmount(request)
     if (!hop) return null
 
-    // If we know a local route to the destinationAccount, use the local route.
+    // note the confusion between `next`, `destination`, and `final`;
+    // `finalAmount` will be renamed to `destinationAmount` later in
+    // the hopToQuote function.
+    if (hop.sourceAmount && hop.finalAmount) {
```

Comment:
Given the curve `[ [0,0], [10, 5] ]`, if you quote by `sourceAmount=1`, the `finalAmount` would be `0` (it's actually `0.5`, but the destination amount gets rounded down).